### PR TITLE
Correct UID template resolution

### PR DESCRIPTION
**Pull request by portante**
Backport of PRs #3029 (665132a89) and #3035 (4822f75af).

When a template key maps to a non-string value, an error is encountered like:

    "TypeError: decoding to str: need a bytes-like object, float found"

The code now converts all non-string values to strings, and ignores any keys which don't map to a value with one of the types: `string`, `int`, or `float`.

A set of unit tests for this operation is provided, and the function name was renamed to ensure its intended use is clear.

Unfortunately, as the bits have aged the backport of the unit tests is a bit more difficult.  To make this all work we had to:

 * Remove fallback `elasticsearch` import, it just confused things

 * Move `pytest` invocation under `jenkins-unittests` since we can't use `--current-env` with `py3-server`

 * Fix `PYTHONPATH` for unit tests to include the server code

 * Delineate the CI steps a bit more cleanly

### CODE REVIEW

**Review by dbutenhof — APPROVED** [review]

**Review by webbnh — APPROVED** [review]
Looks OK, but we're missing a few test cases.

**Comment by webbnh** [comment]
And, of course, there's that problem with the tests....
```
ImportError while importing test module '/home/pbench/server/lib/pbench/test/unit/test_indexer.py'.
...
Traceback:
server/lib/pbench/indexer.py:35: in <module>
    from elasticsearch1 import Elasticsearch, helpers, exceptions as es_excs
E   ModuleNotFoundError: No module named 'elasticsearch1'

During handling of the above exception, another exception occurred:
...
E   ImportError: cannot import name 'decodestring' from 'base64' (/usr/lib64/python3.9/base64.py)
```
Sounds like a dependency problem...but I'm not sure whose....

I just started a re-build of `main`...we'll see if the problem crops up there.

**Comment by webbnh** [comment]
> I just started a re-build of `main`...we'll see if the problem crops up there.

`main` appears to be running fine on [intlab-006](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/manage/computer/intlab-006/).  This PR's build ran on [intlab-004](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/manage/computer/intlab-004/)...I don't know if there is an environmental difference.

**Comment by portante** [comment]
> > I just started a re-build of `main`...we'll see if the problem crops up there.
> 
> `main` appears to be running fine on [intlab-006](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/manage/computer/intlab-006/). This PR's build ran on [intlab-004](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/manage/computer/intlab-004/)...I don't know if there is an environmental difference.

Well this is fortunate.  This happens to be an image build problem where for the `b0.69` branch we are installing the latest `python3-elasticsearch` and then requiring `elasticsearch1`. The pytests are somehow see the `elasticsearch` Python 3 module and not trying to install the `elasticsearch1` requirement.  Not sure how that is happening yet.

File: server/lib/pbench/test/unit/test_indexer.py
Comment (portante) [reply server/lib/pbench/test/unit/test_indexer.py:31]: Yes, those are good tests and I should update them on `main` first.

Comment (portante) [reply server/lib/pbench/test/unit/test_indexer.py:31]: See PR #3035.

**Comment by portante** [comment]
Waiting for PR #3035 to merge before we add those additional tests as part of this back-port.

File: server/lib/pbench/test/unit/test_indexer.py
Comment (portante) [reply server/lib/pbench/test/unit/test_indexer.py]: PR #3035 was merged, and the changes noted above added here.

**Review by portante — COMMENTED** [review]
Ready for review, though this will need a pre-squash to drop the `branch.name` change.

File: jenkins/Pipeline.gy
Comment (webbnh) [inline jenkins/Pipeline.gy:7]: If we're going to do a Pbench Server 0.69 update, shouldn't we have unit testing for it?

**Review by portante — COMMENTED** [review]
Okay, ready for final review, `branch.name` change removed and container image pushed for this branch.

File: jenkins/run-unittests
Comment (webbnh) [inline jenkins/run-unittests:34]: So...does this replace the removal from the `tox` command line?

**Review by dbutenhof — DISMISSED** [review]

**Review by webbnh — APPROVED** [review]
👍

File: jenkins/Pipeline.gy
Comment (portante) [reply jenkins/Pipeline.gy:7]: Yes, we do, see `jenkins-unittests`.

File: jenkins/run-unittests
Comment (portante) [reply jenkins/run-unittests:34]: Yes, see the commit message, we can't use `py3-server` with `--current-env`.

Comment (portante) [reply jenkins/run-unittests:34]: I'll update the PR description as well.

**Review by dbutenhof — APPROVED** [review]

**Review by webbnh — APPROVED** [review]
> Sorry, had to update the commit message.

Couldn't do that via the Squash-and-Merge button, eh?

**Review by ndokos — APPROVED** [review]

**Pull request merged**
